src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -0,0 +1,618 @@
+
+As an example, assume a kernel name is foo having 3 CUs foo_1, foo_2, foo_3. The CUs foo_1 and foo_2 are connected to DDR bank 0, but the CU foo_3 is connected to DDR bank 1. 
+
+       - Opening kernel handle for foo_1 and foo_2 (as they have identical interface connection)
+       
+         .. code:: c++
+               :number-lines: 35
+                  
+                    cu_group_1 = xrtPLKernelOpen(device, xclbin_uuid, "foo:{foo_1,foo_2}");     
+   
+       - Opening kernel handle for foo_3
+          
+         .. code:: c++
+               :number-lines: 35
+                  
+                    cu_group_2 = xrtPLKernelOpen(device, xclbin_uuid, "foo:{foo_3}");     
+
+
+
+**C++**: In C++, xrt::kernel object can be created from the constructor of ``xrt::kernel`` class. 
+
+.. code:: c++
+      :number-lines: 35
+          
+           auto xclbin_uuid = device.load_xclbin("kernel.xclbin");
+           auto krnl = xrt::kernel(device, name, xclbin_uuid); 
+      
+Exclusive access of the kernel's CU
+***********************************
+  
+The API ``xrtPLKernelOpen`` opens a kernel's CU in a shared mode so that the CU can be shared with the other processes. In some cases, it is required to open the CU in exclusive mode (for example, when it is required to read/write CU mapped register). Exclusive CU opening fails if the CU is already opened in either shared or exclusive access. 
+
+.. code:: c
+      :number-lines: 39
+     
+           xrtKernelHandle kernel = xrtPLKernelOpenExclusive(device, xclbin_uuid, "name");
+
+**C++**: When the ``xrt::kernel`` constructor is called with an additional boolean argument set as true, it opens CU in exclusive mode and returns the kernel object.    
+
+.. code:: c++
+      :number-lines: 39
+       
+           auto krnl = xrt::kernel(device, name, xclbin_uuid, true); 
+
+   
+
+Getting bank group index of the kernel argument
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We have seen in the Buffer creation section that it is required to provide the buffer location during the buffer creation. XRT provides an API ``xrtKernelArgGroupId`` (C++: ``xrt::kernel::group_id``) that returns the bank index (ID) of a specific argument of the kernel. This ID is used as the last argument of ``xclAllocBO`` (in C++ with ``xrt::bo`` constructor) API to create the buffer on the same memory bank. 
+
+
+Let us review the example below where the buffer is allocated for the kernel's first (argument index 0) by using this API
+
+.. code:: c
+      :number-lines: 39
+           
+           xrtMemoryGroup idx_0 = xrtKernelArgGroupId(kernel, 0); // bank index of 0th argument
+           xrtBufferHandle a = xrtBOAlloc(device, data_size*sizeof(int), XCL_BO_FLAGS_NONE, idx_0);
+
+
+.. code:: c++
+      :number-lines: 15
+                       
+           auto input_buffer = xrt::bo(device, buffer_size_in_bytes, XCL_BO_FLAGS_NONE, kernel.group_id(0));
```

Comment:
```suggestion
           auto input_buffer = xrt::bo(device, buffer_size_in_bytes, kernel.group_id(0));
```
please provide an overload without useless flag. Or put it after with an optional defaulted value?

Reply:
This PR is not about coding.  Point is valid indeed, but not for this PR.

Reply:
implementation specific suggestion. I am tracking it through a CR, not related to this pull request